File: aexample/src/main/java/org/judal/examples/java/model/StudentCourse.java
package org.judal.examples.java.model;

import javax.jdo.JDOException;

import org.judal.storage.EngineFactory;
import org.judal.storage.java.MapRecord;
import org.judal.storage.relational.RelationalDataSource;

public class StudentCourse extends MapRecord {

	private static final long serialVersionUID = 1L;
	
	private static final String TABLE_NAME = "student_x_course";
	
	public StudentCourse() throws JDOException {
		this((RelationalDataSource) EngineFactory.DefaultThreadDataSource.get());
	}

	public StudentCourse(RelationalDataSource dataSource) throws JDOException {
		super(dataSource, TABLE_NAME);
	}

	public int getCourseId() {
		return getInt("id_course");
	}

	public void setCourseId(final int id) {
		put("id_course", id);
	}

	public int getStudentId() {
		return getInt("id_student");
	}

	public void setStudentId(final int id) {
		put("id_student", id);
	}
	
}

File: aexample/src/main/scala/org/judal/examples/scala/model/StudentCourse.scala
package org.judal.examples.scala.model

import org.judal.storage.EngineFactory
import org.judal.storage.scala.MapRecord
import org.judal.storage.relational.RelationalDataSource

class StudentCourse(dataSource: RelationalDataSource) extends MapRecord(dataSource, StudentCourse.TABLE_NAME) {
	
	def this() = this(EngineFactory.getDefaultRelationalDataSource)

	def getCourseId() : Int = getInt("id_course")

	def setCourseId(id: Int) = put("id_course", id)

	def getStudentId() = getInt("id_student")

	def setStudentId(id: Int) = put("id_student", id)
	
}


object StudentCourse {
  val TABLE_NAME = "student_x_course"
}
